### PR TITLE
MIPS: Adding a check for JAL argument in PIC mode to prevent crashing on getExpr

### PR DESCRIPTION
**Pull request by parsifal-47**
Only an Expr is supported by JAL.
Let's check it before getExpr.
https://github.com/llvm/llvm-project/issues/80535

### CODE REVIEW

**Comment by parsifal-47** [comment]
@wzssyqa could you please take a look?

File: llvm/lib/Target/Mips/AsmParser/MipsAsmParser.cpp [hunk 1]
```diff
@@ -2108,6 +2108,10 @@ bool MipsAsmParser::processInstruction(MCInst &Inst, SMLoc IDLoc,
   if ((Opcode == Mips::JAL || Opcode == Mips::JAL_MM) && inPicMode()) {
     warnIfNoMacro(IDLoc);
 
+    if (!Inst.getOperand(0).isExpr()) {
+      return Error(IDLoc, "jal needs a symbolic expression in PIC mode");
```
Comment (wzssyqa) [inline llvm/lib/Target/Mips/AsmParser/MipsAsmParser.cpp:2112]: gas gives the error message like:
```
k.s: Assembler messages:
k.s:2: Error: unsupported constant in relocation
```
I think that we should keep them same.

Comment (parsifal-47) [reply llvm/lib/Target/Mips/AsmParser/MipsAsmParser.cpp]: sounds good, updated to the same message, thank you!

**Comment by parsifal-47** [comment]
@wzssyqa please take a look, I should have resolved your comment, thank you!

**Comment by wzssyqa** [comment]
Thanks for your work.

**Pull request merged**